lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -497,51 +498,42 @@ export default class BitcoinProcessor {
   }
 
   /**
-   * Begins to revert databases until consistent with blockchain, returns last good height
-   * @returns last valid block height before the fork
+   * Begins to revert databases until consistent with blockchain.
+   * @returns A known valid block before the fork. `undefined` if no known valid block can be found.
    */
-  private async revertDatabases (): Promise<IBlockInfo> {
+  private async revertDatabases (): Promise<IBlockInfo | undefined> {
```

Comment:
I think if we change the contract to say that the function returns then we don't have to return undefined.

> A known valid block before the fork. genesisBlock if no known valid block can be found.

Reply:
Synced up offline and took a valid suggestion of not assigning `this.lastProcessedBlock` to undefined.